download_from_csv.py

```diff
@@ -98,16 +100,27 @@ def download_finish(d):
         'logger': MyLogger(),
         'outtmpl': folder + '/' + song['name'] + ' - ' + song['artist'] + '.%(ext)s'
     }
-    url = ' '.join([song['name'], song['artist'], 'audio', 'youtube'])
+    base_query = ' '.join(song['name'] + song['artist'])
+    if args.query is None:
+        url = '%s %s' % (base_query, 'youtube')
```

Comment:
I found out that appending "audio youtube" results into some weird gvsearch1 issues and the download failed. Issues were solved when I changed it to just "youtube" . Try the following playlist : http://ix.io/p7E if you need to test this.